builtin/logical/pki/path_acme_order.go

```diff
@@ -284,6 +285,18 @@ func (b *backend) acmeFinalizeOrderHandler(ac *acmeContext, _ *logical.Request,
 		return nil, fmt.Errorf("failed saving updated order: %w", err)
 	}
 
+	if account.MaxCertExpiry.Before(order.CertificateExpiry) {
+		b.acmeAccountLock.RLock() // Prevents Account Updates and Tidy Interfering
+		defer b.acmeAccountLock.RUnlock()
+
+		account.MaxCertExpiry = order.CertificateExpiry
+		if ignoreErr := b.acmeState.UpdateAccount(ac, account); ignoreErr != nil {
+			// we shouldn't fail everything if we simply fail to update the timestamp
+			// on the account, it's not a huge deal if we lose this update.
```

Comment:
is it okay if the account is removed by tidy, because MaxCertExpiry was way in the past (and not updated?)

Reply:
No it wouldn't be great, I'll move this UpdateAccount earlier. The main reason I had ignored it was as we had already generated the certificate and updated the order this would be pointless to fail now.

Reply:
Thinking about this a little more, instead of updating the account on order creation like I'm doing here, what if we move this logic into the tidy operation that is already updating the account and holding down the account update lock?

Reply:
Well we only clean up accounts in tidy if we successfully deleted all orders/have none. So there will always be a small race there no matter what I believe locking or not. That is why we are buffering based on account creation to try and avoid this. 

So I don't really see a difference between updating the account MaxCertExpiry here, or within tidy where we are already iterating over the account's orders and have the write lock for the account to set the field there instead. 

The main advantage would be no account locking in the order creation path.

Or have I overlooked something?

Reply:
Ah, gotcha. No, I think that's fine I think. Grabbing the lock here doesn't fix the race I don't believe, so either approach is fine by me.